tests/applications/flag_control_mixin_test.py

```diff
@@ -16,16 +28,18 @@ def test_start_twice():
     app = DummyApp()
     app.start()
     assert app._started and not app._paused
-    # app.start()
-    # assert app._started and not app._paused
+    with pytest.raises(ApplicationStartedException):
+        app.start()
+        assert app._started and not app._paused
```

Comment:
I think (?) this line is never reached, it exists the context as soon as the error is raised.
The checks on the flags should be outside.

Reply:
Oh.. you are right...! I fixed it...! Thank you